app/src/main/java/com/android/sample/model/workout/WorkoutViewModel.kt

```diff
@@ -1,19 +1,44 @@
 package com.android.sample.model.workout
 
 import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
 import kotlinx.coroutines.flow.*
+import kotlinx.coroutines.launch
 
-open class WorkoutViewModel<out T : Workout>(private val repository: WorkoutRepository<T>) :
-    ViewModel() {
+open class WorkoutViewModel<out T : Workout>(
+    private val repository: WorkoutRepository<T>,
+    private val localCache: WorkoutLocalCache,
+    private val workoutClass: Class<T> // Add a class reference for T
+) : ViewModel() {
 
-  val workouts_ = MutableStateFlow<List<@UnsafeVariance T>>(emptyList())
-  val workouts: StateFlow<List<T>> = workouts_
+  val _workouts = MutableStateFlow<List<@UnsafeVariance T>>(emptyList())
+  val workouts: StateFlow<List<T>> = _workouts.asStateFlow()
 
   private val selectedWorkout_ = MutableStateFlow<T?>(null)
-  open val selectedWorkout: StateFlow<T?> = selectedWorkout_
+  open val selectedWorkout: StateFlow<T?> = selectedWorkout_.asStateFlow()
 
   init {
-    repository.init { getWorkouts() }
+    loadCachedWorkouts()
+  }
+
+  private fun loadCachedWorkouts() {
```

Comment:
Is this method necessary as you defined the get documents in the repo so that it first checks for the cache in workoutRepositoryFirestore. It is a bit duplicated.